runtime/src/iree/modules/hal/module.c

```diff
@@ -32,8 +32,8 @@
 // Module type definitions
 //===----------------------------------------------------------------------===//
 
-#define IREE_HAL_MODULE_VERSION_0_3 0x00000003u
-#define IREE_HAL_MODULE_VERSION_LATEST IREE_HAL_MODULE_VERSION_0_3
+#define IREE_HAL_MODULE_VERSION_0_4 0x00000004u
+#define IREE_HAL_MODULE_VERSION_LATEST IREE_HAL_MODULE_VERSION_0_4
```

Comment:
> Future changes will start to add support on targets for the simplified bindings and then remove the existing pipeline layout-based binding model as a breaking ABI change.

How do you plan on sequencing that?

* Module version 4 just adds new functions so it can work with module version 3?
* Module version 5 replaces the old functions with the new ones and can only work with module version 5?

Reply:
good q - since the flag is off by default and no one should be producing binaries requiring v4 I'll probably just reuse that and strip the types/functions at the same time the flag is set. I haven't tried yet but I believe there's no breakage here as only the new methods require the minimum version bump. One day it'd be nice to have a compatibility checking bot (nightly/etc) that tried to load older vmfbs for as long as we claim to support them.

Reply:
> One day it'd be nice to have a compatibility checking bot (nightly/etc) that tried to load older vmfbs for as long as we claim to support them.

Filed https://github.com/iree-org/iree-test-suites/issues/7